partner_nace/wizard/nace_import.py

```diff
@@ -0,0 +1,282 @@
+                    else:
+                        value = n.text
+                    if field_type == 'integer':
+                        try:
+                            value = int(value)
+                        except:
+                            value = 0
+                else:
+                    logger.debug("xpath = '%s', not found" % field_xpath)
+                if field_required and not value:
+                    raise Warning(
+                        _('Value not found for mandatory field %s' % k))
+                item[k] = value
+        return item
+
+    def _download_nace(self, lang_code):
+        url_base = 'http://ec.europa.eu'
+        url_path = '/eurostat/ramon/nomenclatures/index.cfm'
+        url_params = {
+            'TargetUrl': 'ACT_OTH_CLS_DLD',
+            'StrNom': 'NACE_REV2',
+            'StrFormat': 'XML',
+            'StrLanguageCode': lang_code,
+            # 'IntKey': '',
+            # 'IntLevel': '',
+            # 'TxtDelimiter': ';',
+            # 'bExport': '',
+        }
+        url = url_base + url_path + '?'
+        url += '&'.join([k + '=' + v for k, v in url_params.iteritems()])
+        logger.info('Starting to download %s' % url)
+        try:
+            res_request = requests.get(url)
+        except Exception, e:
+            raise Warning(
+                _('Got an error when trying to download the file: %s.') %
+                str(e))
+        if res_request.status_code != requests.codes.ok:
+            raise Warning(
+                _('Got an error %d when trying to download the file %s.')
+                % (res_request.status_code, url))
+        logger.info('Download successfully %d bytes' %
+                    len(res_request.content))
+        # Workaround XML: Remove all characters before <?xml
+        pattern = re.compile(r'^.*<\?xml', re.DOTALL)
+        content_fixed = re.sub(pattern, '<?xml', res_request.content)
+        if not re.match(r'<\?xml', content_fixed):
+            raise Warning(_('Downloaded file is not a valid XML file'))
+        return content_fixed
+
+    @api.model
+    def create_or_update_nace(self, node):
+        if not self._check_node(node):
+            return False
+
+        nace_model = self.env['res.partner.nace']
+        data = self._mapping(node, False)
+        data.update(self._mapping(node, True))
+        level = data.get('level', 0)
+        if level >= 2 and level <= 5:
+            data['parent_id'] = self._parents[level - 2]
+        nace = nace_model.search([('level', '=', data['level']),
+                                  ('code', '=', data['code'])])
+        if nace:
+            nace.write(data)
+        else:
+            nace = nace_model.create(data)
+        if level >= 1 and level <= 4:
+            self._parents[level - 1] = nace.id
+        return nace
+
+    @api.model
+    def translate_nace(self, node, lang):
+        translation_model = self.env['ir.translation']
```

Comment:
OK, I didn't known. In next revision we could change this. Now, this implementation is fully tested and running well.